From 8d3cd37e6ebd5a5bf17cf2fa8d99d0ab811732bf Mon Sep 17 00:00:00 2001
From: Venkatasaiprudhvi Raj Sannidhi <vsannidhi@maxlinear.com>
Date: Sun, 1 Mar 2026 22:06:41 -0800
Subject: [PATCH] WLANRTSYS-102204 [Airties] [internal] Observed Hostapd crash
 during 6GHz test run

Issue: While running Airties tests, when 6G VAPs are enabled/disabled
continuously, the hostapd crash is seen.

Root cause: It is observed that the crash is occuring while bringing
down the non Tx VAP while fetching the sequence number from Tx VAP.
But Tx VAP would already be freed at the driver level and hence the
crash.

Fix: Check the driver interface status before fetching the sequence
number to prevent the crash.

Fixup to: WLANRTSYS-87381 Porting MBSSID changes to wav700

Signed-off-by: Venkatasaiprudhvi Raj Sannidhi <vsannidhi@maxlinear.com>
---
 src/ap/wpa_auth_glue.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/ap/wpa_auth_glue.c b/src/ap/wpa_auth_glue.c
index f2e90dada..b958c5a47 100644
--- a/src/ap/wpa_auth_glue.c
+++ b/src/ap/wpa_auth_glue.c
@@ -722,6 +722,12 @@ static int hostapd_wpa_auth_get_seqnum(void *ctx, const u8 *addr, int idx,
 	struct hostapd_data *hapd = ctx;
 	int link_id = -1;
 
+#ifdef CONFIG_VENDOR_MXL
+	if (!hapd->drv_priv) {
+		wpa_printf(MSG_ERROR, "%s: No driver interface", __func__);
+		return -1;
+	}
+#endif /* CONFIG_VENDOR_MXL */
 #ifdef CONFIG_IEEE80211BE
 	if (hapd->conf->mld_ap && idx)
 		link_id = hapd->mld_link_id;
-- 
2.43.0

